### PR TITLE
Fix "first password email" condition and add tests for user notification upon enrollment in course

### PR DESCRIPTION
**Pull request by stompGT**

### CODE REVIEW

File: spec/services/courses/add_students_service_spec.rb [hunk 1]
```diff
@@ -127,5 +127,30 @@
         expect(current_email).to eq(nil)
       end
     end
+
+    context 'notification is enabled' do
+      let(:notify) { true }
+      let(:school) { course.school }
+
+      it 'send email to user set his first password' do
+        service = double('Users::MailFirstPasswordTokenService')
```
Comment (mpraglowski) [inline spec/services/courses/add_students_service_spec.rb]: instance_double over double

File: spec/services/courses/add_students_service_spec.rb [hunk 2]
```diff
@@ -127,5 +127,30 @@
...
+        allow(Users::MailFirstPasswordTokenService).to receive(:new) { service }
```
Comment (mpraglowski) [inline spec/services/courses/add_students_service_spec.rb]: This is the code we ends up with when there is no proper dependency injection. But let's leave it for now. I want this service to be taken out of this class.

File: spec/services/courses/add_students_service_spec.rb [hunk 3]
```diff
@@ -127,5 +127,30 @@
...
+        expect(service).to receive(:execute)
+
+        subject.add([student_1_data])
```
Comment (mpraglowski) [inline spec/services/courses/add_students_service_spec.rb:140]: This test just test if the service double is executed. This is not really an integration test. Will not allow me for fearless refactoring of this feature :(

Comment (stompGT) [reply spec/services/courses/add_students_service_spec.rb:140]: Now i see the issue to test without dependency injection.
But as it's planned to remove this mail service, would it be more reassuring to just add an expectation to the service initialization?

File: spec/services/courses/add_students_service_spec.rb (hunk 1 above)
Comment (stompGT) [reply spec/services/courses/add_students_service_spec.rb]: Fixed

**Comment by mpraglowski** [comment]
outdated and unclear

**Pull request closed without merging**